arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Validated.kt

```diff
@@ -352,28 +353,32 @@ public sealed class Validated<out E, out A> {
     fold(::Valid, ::Invalid)
 }
 
+public inline fun <E, A, B> Validated<E, A>.zip(combineError: (E, E) -> E, fb: Validated<E, B>): Validated<E, Pair<A, B>> =
+  zip(combineError, fb, ::Pair)
+
+public inline fun <E, A, B, Z> Validated<E, A>.zip(combineError: (E, E) -> E, fb: Validated<E, B>, f: (A, B) -> Z): Validated<E, Z> =
+  when (this) {
+    is Invalid -> when (fb) {
+      is Invalid -> Invalid(combineError(this.value, fb.value))
+      is Valid -> Invalid(this.value)
+    }
+    is Valid -> when (fb) {
+      is Invalid -> Invalid(fb.value)
+      is Valid -> Valid(f(this.value, fb.value))
+    }
+  }
+
+@Deprecated(SemigroupDeprecation, ReplaceWith("SE.run { zip({ x, y -> x + y }, fb, ::Pair) }"))
```

Comment:
I've deprecated both `Semigroup` and `zip` at the same time in the `Validated` PR with `    "Either.zipOrAccumulate({ a, b -> SE.run<Semigroup<E>, E> { a.combine(b) } }, toEither(), fb.toEither(), ::Pair).toValidated()"`. So we probably shouldn't add these new methods.

Reply:
Removed in the merge commit